### ISSUE TITLE
jump to candidate with 'enter' in saved helm-grep buffer results in "Wrong type argument : stringp, nil"

### ISSUE THREAD

**Issue by dbldippy** [youngker/helm-codesearch.el#4]
This package is the best Emacs codesearch integration I have used so far. Far better than my own! 

One thing I often do is saving the result of a codesearch for later reference. The results of helm-codesearch can be successfully saved into a helm-grep buffer with C-x C-s. However, the content of the resulting buffer seems not to be fully compatible with helm-grep, because you cannot jump to a candidate with 'enter'. The above error occurs. 

Also, it would be great if the codesearch content of the elm-grep buffer would be arranged in a way that a search and replace across the results with wgrep-helm's C-c C-p, C-c C-e would work.

To reproduce the error, do:
1. Execute a codesearch with helm-codesearch-find-pattern
2. Save the result as helm-grep buffer with C-x C-s
3. Jump to a candidate with 'enter'

Emacs 25.1.1 on Windows 7
helm-codesearch 20160831.1742
helm 20161019.2251
wgrep-helm 20140528.1427

**Comment by youngker** [comment]
hi wik3
I'm reproducing this issue. 
The cause is helm-grep package.

C-x C-s (wgrep) or F3 (Save results in grep buffer) used a `helm-grep-save-results-1`
and `helm-grep-action` will find below regexp.
`(defvar helm-grep-split-line-regexp "^\\([[:lower:][:upper:]]?:?.*?\\):\\([0-9]+\\):\\(.*\\)")`.

I can not fix this issue. sorry.

If you really want this feature. please use this patch.

``` elisp
                (abbrev-file (format "\n%s" (helm-codesearch-abbreviate-file file)))
                (fake-file (propertize abbrev-file 'helm-candidate-separator t)))
-    (if (string= file helm-codesearch-file)
-        (list (cons display-line candidate))
-      (progn
-        (setq helm-codesearch-file file)
-        (list (cons (format "%s\n%s" fake-file display-line) candidate))))))
+    (list (cons (format "%s:%s: %s" file lineno source) candidate))
+    ;; (if (string= file helm-codesearch-file)
+    ;;     (list (cons display-line candidate))
+    ;;   (progn
+    ;;     (setq helm-codesearch-file file)
+    ;;     (list (cons (format "%s\n%s" fake-file display-line) candidate))))
+    ))

 (defun helm-codesearch-find-pattern-transformer (candidates source)
```

Thanks

**Comment by youngker** [comment]
hi wik3

It has been fixed in version 0.4.0

**Comment by dbldippy** [comment]
Hi Youngker,

Thank you! Working with your mode everyday on major industrial project. Great work. Indispensable.

Mike
